### ISSUE TITLE
Maven Plugin - configurationFile merging

### ISSUE THREAD

**Issue by martinlau** [jOOQ/jOOQ#7143]
Currently the use of `<configurationFile>` is an all or nothing replacement for `<jdbc>` and `<generator>` defined in my pom.xml.

For `<generator>` this is fine (more than that - it's desirable) as I want to manage my generation configuration along side my source code (since it generates source code).

But, `<jdbc>` is a build concern, and so really wants to be defined and managed in my pom.xml.

A bit fo an example: we use the fabric8 docker-maven-plugin to start a DB in container, with the IP / port of the container exposed as maven properties (the following snippet exposes `${db.ip}` and `${db.port}`):

```xml
<plugin>
  <groupId>io.fabric8</groupId>
  <artifactId>docker-maven-plugin</artifactId>
  <version>0.23.0</version>
  <configuration>
    <images>
      <image>
	<name>postgres:10-alpine</name>
	<run>
	  <ports>
	    <port>+db.ip:db.port:5432</port>
          </ports>
```

Ideally, I'd then need to pass to jooq-codegen-maven:

```xml
<plugin>
  <groupId>org.jooq</groupId>
  <artifactId>jooq-codegen-maven</artifactId>
  <version>3.10.4</version>
  <configuration>
    <configurationFile>${project.basedir}/src/build/jooq.xml</configurationFile>
    <logging>DEBUG</logging>
    <jdbc>
      <user>${db.user}</user>
      <password>${db.password}</password>
      <url>jdbc:postgresql://${db.ip}:${db.port}/${db.name}</url>
    </jdbc>
  </configuration>
```

But this doesn't work since `<configurationFile>` replaces both `<jdbc>` and `<generator>` ([Plugin.java:136](https://github.com/jOOQ/jOOQ/blob/version-3.10.4/jOOQ-codegen-maven/src/main/java/org/jooq/util/maven/Plugin.java#L136)).

I don't know if a straight null check is enough:

```java
if (configuration.getGenerator() != null) {
  generator = configuration.getGenerator();
}
if (configuration.getJdbc() != null) {
  jdbc = configuration.getJdbc();
}
```

... or whether there's a world of pain in implementing this trying to distinguish between a default configuration and a real one?

**Comment by lukaseder** [comment]
> I don't know if a straight null check is enough

You're probably right, that might be enough. I'll look into this. There had been other requests to merge different Maven configurations in the past, such as #6556 (although not the external one with the inline one). So, I definitely think this could be something that people would like to see working.

**Comment by lukaseder** [comment]
This has been implemented in jOOQ 3.12 through #6548 (Support multiple external configuration files through <configurationFiles/>)
